### PR TITLE
Adding spark example method

### PR DESCRIPTION
**Pull request by ffinfo**
### Description

_Give your PR a **concise** yet **descriptive** title_

----

### Checklist (never delete this)

- [ ] Each method/class/object/trait should have scaladocs
- [ ] Added methods are unit tested
- [ ] Test coverage may not drop
- [ ] Documentation should be updated if required

### CODE REVIEW

File: src/main/scala/nl/biopet/utils/tool/ToolCommand.scala [hunk 1]
```diff
@@ -145,19 +145,34 @@ trait ToolCommand[Args] extends Logging {
...
+  }
+
+  /** Convert and *not* tests args */
+  def sparkUnsafeExample(args: String*): String = {
+    exampleToMarkdown(true, args: _*)
   }
 
   /** Common function to convert to string */
-  private def exampleToMarkdown(args: String*): String = {
+  private def exampleToMarkdown(spark: Boolean, args: String*): String = {
```
Comment (rhpvorderman) [inline src/main/scala/nl/biopet/utils/tool/ToolCommand.scala:176]: I do not like that this spark value is required. It's better to use 
```Scala
(args: String*, spark: Boolean = false)
```
instead.

File: src/main/scala/nl/biopet/utils/tool/ToolCommand.scala [hunk 2]
```diff
@@ -145,19 +145,34 @@ trait ToolCommand[Args] extends Logging {
   def example(args: String*): String = {
     cmdArrayToArgs(args.toArray)
 
-    exampleToMarkdown(args: _*)
+    exampleToMarkdown(false, args: _*)
```
Comment (rhpvorderman) [inline src/main/scala/nl/biopet/utils/tool/ToolCommand.scala]: See comment above. I would prefer
```Scala
exampleToMarkdown(args: _*)
``` 
here

File: src/main/scala/nl/biopet/utils/tool/ToolCommand.scala [hunk 3]
```diff
@@ -145,19 +145,34 @@ trait ToolCommand[Args] extends Logging {
...
   def unsafeExample(args: String*): String = {
-    exampleToMarkdown(args: _*)
+    exampleToMarkdown(false, args: _*)
+  }
+
+  /** Convert and tests args */
+  def sparkExample(args: String*): String = {
+    cmdArrayToArgs(args.toArray)
+
+    exampleToMarkdown(true, args: _*)
```
Comment (rhpvorderman) [inline src/main/scala/nl/biopet/utils/tool/ToolCommand.scala]: I would prefer
```Scala
exampleToMarkdown(args: _*, spark=true)
```
here

**Review by rhpvorderman — CHANGES_REQUESTED** [review]

File: src/main/scala/nl/biopet/utils/tool/ToolCommand.scala (hunk 1 above)
Comment (ffinfo) [reply src/main/scala/nl/biopet/utils/tool/ToolCommand.scala:176]: Not possible, * arguments must be the last

File: src/main/scala/nl/biopet/utils/tool/ToolCommand.scala (hunk 2 above)
Comment (ffinfo) [reply src/main/scala/nl/biopet/utils/tool/ToolCommand.scala]: also not possible ;)

File: src/main/scala/nl/biopet/utils/tool/ToolCommand.scala (hunk 3 above)
Comment (ffinfo) [reply src/main/scala/nl/biopet/utils/tool/ToolCommand.scala]: also not possible ;)

**Pull request merged**